### DISCUSSION TITLE
Did Streamlink handle Twitch hosts in the past?

### DISCUSSION THREAD

**Discussion by lewisje** [streamlink/streamlink#5102] [Q&A] [answered]
For the past few years, I've only used Streamlink together with Chatty, to watch Twitch streams; in October, Twitch removed channel-hosting, and raids started working differently, rather than including hosts.

A few days ago, a new stable version of Chatty started listening for raids and notifying about them (because of API limitations, only 10 of the channels one is watching can be listened to at a time), and I noticed that even though it notifies about raids in chat again, the stream does not switch over as it did when hosts still existed.

I didn't notice anything in the `master` branch of either program that looked like it was specifically about hosts or raids, so I asked a question in [the Chatty issue about the topic](https://github.com/chatty/chatty/issues/398#issuecomment-1397375232) and I'm not sure about whether I should bother generating a debug log and waiting for another raid, which I'd only do if host-following were something that was done on Streamlink's side to begin with (rather than Chatty asking Streamlink to change the stream URL).

To put it another way, if it *wasn't* something that had been done on Chatty's side in the past, I'd file a bug report for the Twitch plugin, and otherwise, I'd put in a feature request if Chatty for whatever reason decides to stop doing it.

---
I can't directly find an answer to my question with Streamlink from the command line anymore, because channel-hosting is gone, although I *could* see whether it tries and fails to follow raids.

**Comment by bastimeyer** [comment, marked as answer]
- https://github.com/streamlink/streamlink/pull/4805
- https://github.com/streamlink/streamlink/commit/9923b40a6e79480909518bb37f5db238ad9b08ed

> the stream does not switch over
> follow raids

Streamlink doesn't work like this. It can't output two separate streams after another. You will have to restart with the new channel. The old hosting implementation simply did a redirection when trying to launch a stream from a channel which was hosting another one. "Twitch raids" are just a redirection when a stream ends. This is completely different.

**Reply by lewisje** [reply]
Thank you for clarifying this.
